app/src/main/kotlin/com/wire/android/ui/home/conversations/sendmessage/SendMessageViewModel.kt

```diff
@@ -294,18 +294,29 @@ class SendMessageViewModel @Inject constructor(
         }
     }
 
-    private fun Either<CoreFailure, Unit>.handleLegalHoldFailureAfterSendingMessage(conversationId: ConversationId) =
+    private fun Either<CoreFailure, Unit>.handleLegalHoldFailureAfterSendingMessage(
+        conversationId: ConversationId
+    ): Either<CoreFailure, Unit> =
         onFailure { it.handleLegalHoldFailureAfterSendingMessage(conversationId) }
 
-    private fun ScheduleNewAssetMessageResult.handleLegalHoldFailureAfterSendingMessage(conversationId: ConversationId) = let {
-        if (it is ScheduleNewAssetMessageResult.Failure) {
-            it.coreFailure.handleLegalHoldFailureAfterSendingMessage(conversationId)
-        }
-        when (this) {
-            is ScheduleNewAssetMessageResult.Failure -> Either.Left(coreFailure)
-            is ScheduleNewAssetMessageResult.Success -> Either.Right(Unit)
+    private fun ScheduleNewAssetMessageResult.handleLegalHoldFailureAfterSendingMessage(
+        conversationId: ConversationId
+    ): Either<CoreFailure?, Unit> =
+        let {
+            when (this) {
+                is ScheduleNewAssetMessageResult.Success -> Either.Right(Unit)
+                ScheduleNewAssetMessageResult.Failure.DisabledByTeam,
+                ScheduleNewAssetMessageResult.Failure.RestrictedFileType -> {
+                    onSnackbarMessage(ConversationSnackbarMessages.ErrorAssetRestriction)
+                    Either.Left(null)
```

Comment:
suggestion: why not returning the error itself as left? instead of returning customised errors?
and show handling the show message in the compose?

Reply:
the consumer of this error does not care about the type, just if the value is Left or Right, can be TBH it can be replaced with simple boolean and it will do the same + it takes only core failure which our errors in that case are not